packages/admin/admin-rte/src/core/xml/translateAndTransformXmlToState.ts

```diff
@@ -0,0 +1,198 @@
+                id: parseInt(id ?? ""),
+                offset,
+                length: 0,
+            });
+        } else {
+            // take top element of stack when closing pseudo tags occurs
+            const openingTag = stylesStack.pop();
+
+            if (openingTag) {
+                // check if style already occured before
+                const existingStyle = newInlineStyleRanges.findIndex((style) => style.id === openingTag.id);
+
+                if (existingStyle !== -1) {
+                    // update range length of style
+                    newInlineStyleRanges[existingStyle] = {
+                        ...newInlineStyleRanges[existingStyle],
+                        length: newInlineStyleRanges[existingStyle].length + (offset - openingTag.offset),
+                    };
+                } else {
+                    newInlineStyleRanges.push({
+                        id: openingTag.id,
+                        offset: openingTag.offset,
+                        length: offset - openingTag.offset,
+                    });
+                }
+            }
+        }
+
+        // update index shift according to pseudo tag length
+        shiftCount += match[0].length;
+    }
+
+    shiftCount = 0;
+
+    // entity ranges work the same way as inline styles
+    for (const match of entityRangesTags) {
+        const id = match[0].match(/\d+/)?.[0];
+
+        if (match.index === null || match.index === undefined) continue;
+
+        const offset = match.index - shiftCount;
+
+        if (id) {
+            entitiesStack.push({
+                key: parseInt(id ?? ""),
+                offset,
+                length: 0,
+            });
+        } else {
+            const openingTag = entitiesStack.pop();
+
+            if (openingTag) {
+                const existingStyle = newEntityRanges.findIndex((style) => style.key === openingTag.key);
+
+                if (existingStyle !== -1) {
+                    newEntityRanges[existingStyle] = {
+                        ...newEntityRanges[existingStyle],
+                        length: newEntityRanges[existingStyle].length + (offset - openingTag.offset),
+                    };
+                } else {
+                    newEntityRanges.push({
+                        key: openingTag.key,
+                        offset: openingTag.offset,
+                        length: offset - openingTag.offset,
+                    });
+                }
+            }
+        }
+
+        shiftCount += match[0].length;
+    }
+
+    /* inline styles need to be sorted by offset to map style property as inline styles are sorted by style order */
+    block.inlineStyleRanges = block.inlineStyleRanges
+        .sort((a, b) => a.offset - b.offset)
+        .map((style, index) => {
+            const newStyle = newInlineStyleRanges.find((newStyle) => newStyle.id - 1 === index);
+
+            if (!newStyle) {
+                return undefined;
+            }
```

Comment:
When may this happen?

Reply:
this is needed to remove entity ranges when they are missing an opening tag